pkg/asset/installconfig/aws/metadata.go

```diff
@@ -33,3 +43,23 @@ func (m *Metadata) Session(ctx context.Context) (*session.Session, error) {
 
 	return m.session, nil
 }
+
+// AvailabilityZones retrieves a list of availability zones for the configured region.
+func (m *Metadata) AvailabilityZones(ctx context.Context) ([]string, error) {
+	m.Lock()
+	defer m.Unlock()
+
+	if len(m.availabilityZones) == 0 {
+		session, err := m.unlockedSession(ctx)
```

Comment:
can't we load the session separately? it already has the cached mechanics.. also helps lock/unlock separately without `unlocked` version.

Reply:
> can't we load the session separately?

Rolling this discussion into [the earlier thread][1].

[1]: https://github.com/openshift/installer/pull/2512#discussion_r334568035